src/private/symlink_package.bzl

```diff
@@ -16,23 +15,22 @@ The test environments will not have the execroot `node_modules`, or user workspa
 """
 
 def _symlink_impl(ctx):
-    destination = ctx.actions.declare_symlink(ctx.attr.name)
-
     src = ctx.attr.src
 
-    if not src.label.name.startswith("node_modules/"):
-        fail("%s is not a linked npm package" % src.label)
+    has_js_info = JsInfo in src
+    if has_js_info:
+        src_dir = src[JsInfo].npm_sources.to_list()[-1]
+    else:
+        src_dir = src[NpmPackageInfo].src
```

Comment:
Should this be gated and checked, with a better failure message if `NpmPackageInfo` is not present either?

Reply:
I'm fine adding it, but the error message is pretty obvious + there is a file stack pointing to here